app-creator/src/widgets/draggable-widget/draggable-widget.ts

```diff
@@ -151,25 +202,29 @@ export class DraggableWidget extends LitElement {
    * @param e dragend event
    */
   handleDragend() {
-    const draggingElement = store.draggingElement;
-    if (draggingElement && store.elementAdded && !store.reordering) {
-      store.widgetIDs[draggingElement.id]++;
+    const draggingWidget = store.getState().draggingWidget;
+    if (
+      draggingWidget &&
+      store.getState().elementAdded &&
+      !store.getState().reordering
```

Comment:
Re: consolidating the state logic comment above.

For example, this check would become a little easier to read:
```
if (
  element &&
  store.getState().eventType === 'adding' // Better, an enum type
 )
```